ygen/genstate.go

```diff
@@ -52,6 +52,9 @@ type MappedType struct {
 	// the generated code from the structs maintains only type validation,
 	// this is not currently a limitation.
 	UnionTypes map[string]int
+	// UnionTypeInfos stores more information about each union subtype that
+	// can be populated if needed during the code generation pass.
+	UnionTypeInfos map[string]MappedUnionSubtype
```

Comment:
```suggestion
	UnionTypeInfo map[string]MappedUnionSubtype
```

nitty naming suggestion.

Reply:
I like having the plural because maps contain multiple units of `UnionTypeInfo`. I know 'info' is an uncountable noun, but I don't like English's exceptions when being used in a programming environment which is more about rules rather than exceptions. What do you think :)?

Reply:
So actually once `UnionTypes` is removed, I'll rename this to this name. How about that?

Reply:
That sounds reasonable to me.